packages/next/server/config.ts

```diff
@@ -637,6 +637,12 @@ function assignDefaults(userConfig: { [key: string]: any }) {
         `Specified i18n.defaultLocale should be included in i18n.locales.\nSee more info here: https://nextjs.org/docs/messages/invalid-i18n-config`
       )
     }
+    
+    if ((new Set(i18n.locales)).size !== i18n.locales.length) {
```

Comment:
```suggestion
    if ((new Set(i18n.locales.map(locale => locale.toLowerCase())).size !== i18n.locales.length) {
```
Should we normalize to lower case before de-duping here?

Reply:
Yes definitely, I have things like `zh-Hans` and I do normalize them in my app during comparisons